src/blocks/scratch3_looks.js

```diff
@@ -344,73 +344,94 @@ class Scratch3LooksBlocks {
+     * @param {!Target} target Target to set costume to.
      * @param {Any} requestedCostume Costume requested, e.g., 0, 'name', etc.
      * @param {boolean=} optZeroIndex Set to zero-index the requestedCostume.
      * @return {Array.<!Thread>} Any threads started by this switch.
      */
-    _setCostumeOrBackdrop (target,
-        requestedCostume, optZeroIndex) {
+    _setCostume (target, requestedCostume, optZeroIndex) {
         if (typeof requestedCostume === 'number') {
-            target.setCostume(optZeroIndex ?
-                requestedCostume : requestedCostume - 1);
+            target.setCostume(optZeroIndex ? requestedCostume : requestedCostume - 1);
         } else {
-            const costumeIndex = target.getCostumeIndexByName(requestedCostume);
-            if (costumeIndex > -1) {
+            const costumeIndex = target.getCostumeIndexByName(requestedCostume.toString());
+
+            if (costumeIndex !== -1) {
                 target.setCostume(costumeIndex);
-            } else if (requestedCostume === 'previous costume' ||
-                       requestedCostume === 'previous backdrop') {
-                target.setCostume(target.currentCostume - 1);
-            } else if (requestedCostume === 'next costume' ||
-                       requestedCostume === 'next backdrop') {
+            } else if (requestedCostume === 'next costume') {
                 target.setCostume(target.currentCostume + 1);
-            } else if (requestedCostume === 'random backdrop') {
-                const numCostumes = target.getCostumes().length;
+            } else if (requestedCostume === 'previous costume') {
+                target.setCostume(target.currentCostume - 1);
+            } else if (!isNaN(requestedCostume) &&
+                        (typeof requestedCostume !== 'string' || /\d/g.test(requestedCostume))) {
+                target.setCostume(optZeroIndex ? Number(requestedCostume) : Number(requestedCostume) - 1);
+            }
+        }
+
+        // Per 2.0, 'switch costume' can't start threads even in the Stage.
+        return [];
+    }
+
+    /**
+     * Utility function to set the backdrop of a target.
+     * Matches the behavior of Scratch 2.0 for different types of arguments.
+     * @param {!Target} stage Target to set backdrop to.
+     * @param {Any} requestedBackdrop Backdrop requested, e.g., 0, 'name', etc.
+     * @param {boolean=} optZeroIndex Set to zero-index the requestedBackdrop.
+     * @return {Array.<!Thread>} Any threads started by this switch.
+     */
+    _setBackdrop (stage, requestedBackdrop, optZeroIndex) {
+        if (typeof requestedBackdrop === 'number') {
+            stage.setCostume(optZeroIndex ? requestedBackdrop : requestedBackdrop - 1);
+        } else if (requestedBackdrop === 'next backdrop') {
+            stage.setCostume(stage.currentCostume + 1);
+        } else if (requestedBackdrop === 'previous backdrop') {
+            stage.setCostume(stage.currentCostume - 1);
```

Comment:
This behavior is consistent with Scratch 2.0, as you noted here: 

https://github.com/LLK/scratch-vm/issues/1490#issuecomment-413714669

But from a usability/consistency perspective this definitely seems like a bug.

My thought is to make backdrops more consistent with costumes in this regard (e.g. if you have a costume named 'next costume' you can switch to it).
@thisandagain do you have any thoughts on this?

At the very least, the `random backdrop` option should be handled the same way we decide to handle `previous backdrop` and `next backdrop`

Reply:
Agreed @kchadha. I think this is a place I'd be happy to have a breaking change. We should document it here: https://github.com/LLK/scratch-vm/wiki/Compatibility